### DISCUSSION TITLE
Create a `Vec` from a `FixedSizeListArray`

### DISCUSSION THREAD

**Discussion by daniellga** [jorgecarleitao/arrow2#1357] [General]
Hi! Thanks for the package!

I am trying to replicate complex numbers in Arrow using a FixedSizeListArray of 2 arrays. The thing is, I need to use `rustfft` crate to get the fft of this `FixedSizeListArray` but `rustfft` only accepts `&mut [Complex<T>]` types as input. So I ended up thinking of converting the `FixedSizeListArray` to a `Vec<Complex<f64>>`. 

- Is creating the `Vec` the way to go or do you suggest something more efficient? I am not very used to Arrow.
- Is the code below the more efficient way to get a `Vec` from a `FixedSizeListArray`?
```
            let arr = Float64Array::from_slice(&[1., 2., 3., 4., 5., 6.]).boxed();
            let field = Box::new(Field::new("item", arr.data_type().clone(), true));
            let data_type = DataType::FixedSizeList(field, 2);
            let arr = FixedSizeListArray::try_new(data_type, arr, None).unwrap();

            let slice = arr
                .values()
                .as_any()
                .downcast_ref::<PrimitiveArray<f64>>()
                .unwrap()
                .values()
                .as_slice();

            let vec: Vec<Complex<f64>> = slice
                .chunks(2)
                .map(|x| Complex<f64>::new(re: x[0], im: x[1]))
                .collect();
```
Thanks!

**Comment by jorgecarleitao** [comment]
Hey! Yes, that would be the way. Do note that this discards nullability information.

Consider using `chunks_exact(2)` instead of `chunks(2)` as `.values` is guaranteed to have a multiple of the fixed size of the FixedSizeList.

Creating a new vector is required because of the `Complex` requirement, which may have a (pointer) alignment incompatible with one in `values`.

**Reply by daniellga** [reply]
Nice! Thanks a lot!
